rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java

```diff
@@ -40,9 +44,24 @@ public RedeemData findRedeemDataFromScriptHash(byte[] payToScriptHash) {
 
             Federation destinationFederationInstance = destinationFederation.get();
             Script fedRedeemScript = destinationFederationInstance.getRedeemScript();
-            Script fastBridgeRedeemScript = FastBridgeRedeemScriptParser
-                .createMultiSigFastBridgeRedeemScript(fedRedeemScript,
-                    Sha256Hash.wrap(fastBridgeFederationInformationInstance.getDerivationHash().getBytes()));
+
+            RedeemScriptParser parser = RedeemScriptParserFactory.get(fedRedeemScript.getChunks());
+            Script fastBridgeRedeemScript;
+
+            if (parser.getMultiSigType().equals(MultiSigType.ERP_FED)) {
```

Comment:
MultiSigType.ERP_FED because of this: `Federation destinationFederationInstance = destinationFederation.get();`
You only may have a Federation or an ErpFederation. Now, if you have an ErpFederation and you have fast bridge data too, so you need to create a FastBridgeErpRedeemScript.